Orange/widgets/utils/state_summary.py

```diff
@@ -0,0 +1,74 @@
+from Orange.data import StringVariable, DiscreteVariable, ContinuousVariable, \
+    TimeVariable
+
+
+def format_variables_string(variables):
+    """
+    A function that formats the descriptive part of the input/output summary for
+    either features, targets or metas of the input dataset.
+
+    :param variables: Features, targets or metas of the input dataset
+    :return: A formatted string
+    """
+    agg = []
+    for var_type_name, var_type in [('categorical', DiscreteVariable),
+                                    ('numeric', ContinuousVariable),
+                                    ('time', TimeVariable),
+                                    ('string', StringVariable)]:
```

Comment:
Put this list into a variable, e.g. `names_types = [('categorical', DiscreteVariable), ...]`, so you can reuse it below. (See the comment below.)